### ISSUE TITLE
Eraser might not work as intended

### ISSUE THREAD

**Issue by romainrichard** [SerGreen/TheWitnessPuzzles#26]
Seed: `000000213`
I don't believe the following path should solve the puzzle:
![Screenshot_2024-06-20-02-10-44-64_b9da37f254079e85909e85e628f7a824](https://github.com/SerGreen/TheWitnessPuzzles/assets/498307/910f5afb-9c33-432a-b222-ee673ec0c619)
Once the pink square is taken care of by the first eraser, the second eraser has nothing left to fix. I don't believe erasing a green square is valid. So that shouldn't work.

A correct solution for this puzzle would be:
![Screenshot_2024-06-20-02-15-01-45_b9da37f254079e85909e85e628f7a824](https://github.com/SerGreen/TheWitnessPuzzles/assets/498307/f2dcfe2b-4364-4aa6-93c0-f4e1db8ea1c4)
In this case the second eraser takes care of the two triangles which needs fixing.

**Comment by SerGreen** [comment]
Yes, i believe you're correct. Eraser must erase an unfulfilled rule and can not erase a satisfied one. More than two rules of one colour may exist in the same sector, unless one of them is a sun shape, but they're not. Green square shouldn't be erased.
